src/main/java/uk/gov/companieshouse/web/accounts/validation/smallfull/DirectorValidator.java

```diff
@@ -81,6 +99,46 @@
             validationErrors.add(error);
         }
 
+        if(addOrRemoveDirectors.getExistingDirectors() != null) {
+
+            ApiClient apiClient = apiClientService.getApiClient();
+
+            SmallFullApi smallFullApi = smallFullService
+                    .getSmallFullAccounts(apiClient, transactionId, companyAccountsId);
+
+            LocalDate periodStartOn = smallFullApi.getNextAccounts().getPeriodStartOn();
+            LocalDate periodEndOn = smallFullApi.getNextAccounts().getPeriodEndOn();
+
+            for (Director director : addOrRemoveDirectors.getExistingDirectors()) {
+
+                if (!isValidAppointmentOrResignationDate(director, periodStartOn, periodEndOn)) {
+                    ValidationError error = new ValidationError();
+                    error.setFieldPath("");
+                    error.setMessageKey(OUTSIDE_VALID_DATE_RANGE);
+                    validationErrors.add(error);
+
+                    break;
+                }
+            }
+        }
+
         return validationErrors;
     }
+
+    private boolean isValidAppointmentOrResignationDate(Director director, LocalDate periodStartOn, LocalDate periodEndOn) {
+        boolean isValid = true;
+
+        if (director.getResignationDate() != null) {
+            if (director.getResignationDate().isBefore(periodStartOn) || director.getResignationDate().isAfter(periodEndOn)) {
```

Comment:
Consider combining this if with the wrapping if?

Reply:
Took this advice, new code pushed contains changes.